core/client.h

```diff
@@ -19,26 +19,31 @@ namespace ycsbc {
 
 inline int ClientThread(ycsbc::DB *db, ycsbc::CoreWorkload *wl, const int num_ops, bool is_loading,
                         bool init_db, bool cleanup_db, CountDownLatch *latch) {
-  if (init_db) {
-    db->Init();
-  }
-
-  int ops = 0;
-  for (int i = 0; i < num_ops; ++i) {
-    if (is_loading) {
-      wl->DoInsert(*db);
-    } else {
-      wl->DoTransaction(*db);
+    try {
+        if (init_db) {
+            db->Init();
+        }
+
+        int ops = 0;
+        for (int i = 0; i < num_ops; ++i) {
+            if (is_loading) {
+                wl->DoInsert(*db);
+            } else {
+                wl->DoTransaction(*db);
+            }
+            ops++;
+        }
+
+        if (cleanup_db) {
+            db->Cleanup();
+        }
+
+        latch->CountDown();
+        return ops;
+    } catch(const utils::Exception& e) {
+        std::cerr<<"Caught exception: "<<e.what()<<std::endl;
+        return -1;
```

Comment:
Catching exception is good. But in this case, other client threads may still be running?

Reply:
Yes, you're right, I didn't realize it before. I see you have fixed it, thank you.